geonode/base/utils.py

```diff
@@ -220,6 +220,9 @@ def _restore_owner_permissions(self):
                     assign_perm(perm, self.resource.owner, self.resource)
 
 
-def make_public(resource):
-    resource.set_default_permissions()
-    resource.clear_dirty_state()
+def add_keywords_to_resource(resource, keywords):
+    for keyword in keywords:
```

Comment:
I'll keep this utility only for tests purposes because, for a normal flow, the `KeywordHandler` has the responsibility to assign the keywords to the resource. 
It converts the keyword to a specific format in order to let the object understand if is normal keywords or thesaurus once. If you are interested, here are the test cases of the keyword handler -> [link](https://github.com/GeoNode/geonode/blob/0c8b6b83afb4271109041fb4d3ca87ed53577eed/geonode/upload/tests/test_utils.py#L69)

Reply:
Our `keywords` are in fact ManyToMany field behind the scene by Taggit. As taggit [document admits it](https://github.com/jazzband/django-taggit/blob/master/taggit/managers.py#59) that if it is not managed by their manager we can do with it all Django goodness. But as I said earlier, if most feel this method to be for test purpose only, I should move it to `populate_test_data` itself.